### ISSUE TITLE
[Features Request] Slang Shader Support.

### ISSUE THREAD

**Issue by RinMaru** [higan-emu/ares#1384]
I'm not entirely sure what slang shaders are but they seem to be the new hot thing on RetroArch and they have some interesting shaders I'd personally love to see ares use. Its hard to find shaders outsides Quarks Shaders.  Since slang seems to be about a cleaner way to do things I'd like this to be supported in Ares. I know its a tough thing to do and if there's anyway i can make a bounty or entice someone to work on this id be great full. 

I only found Ares late 2023 and i was all over playing my games in HD the way they were never intended to be played. But then the community showed me shaders and the unpleasant effects of what playing these in HD brings. They taught me about shaders and i can never go back to sharp pixels i need my CRT-Royal shader. Slang now have more robust shaders and id love to have them in Ares 

I hope to relive my childhood in these shaders onday. If there's a way to make it happen or even put a bounty on it id love to help.

**Comment by LukeUsher** [comment]
There is the possibility of utilising [librashader](https://github.com/SnowflakePowered/librashader) for this, but there are a few things that need to be solved first.

The first caveat is that it is a rust-based library, meaning we can't incorporate it into our build system: we rely on upstream builds for operating system/architecture support, but that might be worth it given the cost/benefit of a standard shader system. ares can be compiled for many non-common architectures, including PowerPC, and RISC-V, these platforms would lose shader support (although if these platforms do not support OpenGL 3.X, they would already have not supported shaders anyway)

If we use librashader in its current form, we would lose support for shaders on macOS, and potentially aarch64 and other non-standard architectures. As long as we can support x64 and aarch64 (Across Windows, macOS and Linux) I would be happy to switch over.

Librashader needs to be updated to support macOS: compiling as a dylib and updating librashader_ld to look for dylib shared libraries, in addition to .dll and .so; currently it is only possible to compile librashader for Windows and Linux; I would be happy with a working OpenGL backend for macOS, as ares does not currently have a metal backend, just OpenGL support is enough for us.

To fully benefit from librashader and allow the use of shaders with DirectX on Windows, we would also need a D3D11 or D3D12 renderer within ares, but as shaders are currently only working with OpenGL, this is an optional requirement.

TLDR: Librashader needs macOS support and aarch64 binary builds before we can incorporate it.

**Comment by RinMaru** [comment]
wouldnt Vulkan be better than DX11 and 12?

**Comment by LukeUsher** [comment]
This is being worked on in the [librashader ](https://github.com/ares-emulator/ares/tree/librashader) branch

**Comment by RinMaru** [comment]
Should i close this or leave it open for further comments?

**Comment by LukeUsher** [comment]
I think we can close this now: experimental support is in nightlies.
